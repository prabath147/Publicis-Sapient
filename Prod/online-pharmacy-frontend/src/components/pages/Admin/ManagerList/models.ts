import { PageableResponse } from "../storeList/models";

export interface Manager 
{
  managerId: number;
  name: string;
  phoneNo: string;
  licenseNo: string;
  registrationDate: Date;
  approvalStatus: "APPROVED" | "PENDING" | "REJECTED";
}


export interface ManagerListResponse extends PageableResponse{
  data:Manager[]
}