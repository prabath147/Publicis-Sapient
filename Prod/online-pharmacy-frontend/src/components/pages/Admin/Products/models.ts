
export interface CategorySet {
  categoryId: number;
  categoryName: string;
}
export interface Product {
  categorySet: CategorySet[];
  description: string;
  dosageForm: string;
  imageUrl: string;
  productId: number;
  productName: string;
  productType: boolean;
  proprietaryName: string;
  quantity: number;
}

