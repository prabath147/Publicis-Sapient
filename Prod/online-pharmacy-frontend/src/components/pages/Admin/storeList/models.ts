import { Item } from "../../Manager/models";

export interface Test {
  data: Store[];
  pageNumber: number;
  pageSize: number;
  totalRecords: number;
  totalPages: number;
  isLastPage: boolean;
}


export interface Store {
  storeId: number;
  storeName: string;
  address: Address;
  createdDate: string;
  manager: Manager;
  revenue: number;
}

export interface Address {
  addressId: number;
  street: string;
  city: string;
  state: string;
  pinCode: number;
  country: string;
}

export interface Manager {
  managerId: number;
  name: string;
  phoneNo: string;
  licenseNo: string;
  registrationDate: string;
  approvalStatus: string;
}

export interface PageableResponse{
  "pageNumber":number,
  "pageSize":number,
  "totalRecords":number,
  "totalPages":number,
  "isLastPage":boolean
}

export interface getStoreInventoryResponse extends PageableResponse{
  data : Item[]
}


