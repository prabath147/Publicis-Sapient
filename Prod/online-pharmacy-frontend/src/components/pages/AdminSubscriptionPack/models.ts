export interface Subscriptionv{
    subscriptionId:number,
    subscriptionName:string,
    description:string,
    startDate:Date,
    endDate:Date,
    benefits:Benefit
    subscriptionCost:number,
    period:number,
    status:string
}

export interface Benefit{
    delivery_discount:number,
    one_day_delivery:boolean
}
export interface SubDetails{
    subscriptionName:string,
    description:string,
    startDate:Date,
    endDate:Date,
    benefits:Benefit,
    subscriptionCost:number,
    period:number
}

export interface SubRequest{
    name:string,
    description:string,
    stDate:Date,
    edDate:Date,
    deliveryDiscount:number,
    oneDayDelivery:string,
    cost:number,
    period:number
}