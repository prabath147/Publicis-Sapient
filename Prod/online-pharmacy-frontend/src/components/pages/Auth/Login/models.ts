export interface User {
  id: number;
  username: string;
  email: string;
  role: string;
}

export interface LoginRequest {
  username: string;
  password: string;
}

export interface LoginResponse {
  id: number;
  username: string;
  email: string;

  token: string;
  refreshToken: string;
}

export interface tokenStructure {
  sub: string;
  role: string;
  iat: number;
  exp: number;
  id:number;
}
