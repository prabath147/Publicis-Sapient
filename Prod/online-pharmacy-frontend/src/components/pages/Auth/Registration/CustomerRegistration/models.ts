import { AddressInterface } from './../../../../ui/forms/AddressForm';

export interface CustomerRegistrationRequest {
  username: string;
  email: string;
  password: string;
  role: ["customer"];

  detailObject: {
    fullName: string;
    mobileNumber: number;
    address: AddressInterface
  };
}

export interface CustomerRegistrationForm {
  fullName: string;
  username: string;
  mobileNumber: number;
  email: string;
  password: string;
  confirmPassword: string;
  address: AddressInterface
}
