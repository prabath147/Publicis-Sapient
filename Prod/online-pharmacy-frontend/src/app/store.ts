import { Action, configureStore, ThunkAction } from "@reduxjs/toolkit";
import StoreReducer from "../components/pages/Admin/storeList/StoreSlice";
import SubscriptionReducer from "../components/pages/AdminSubscriptionPack/SubscriptionvSlice";
import UserReducer from "../components/pages/Auth/Login/UserSlice";
import CartReducer from "../components/pages/Cart/CartSlice";
import ManagerReducer from "../components/pages/Manager/ManagerSlice";
import NotificationReducer from "../components/pages/Open/notificationList/NotificationSlice";
// import OrderSlice from "../components/pages/orderDetail/OrderSlice";
// import OrderListSlice from "../components/pages/orderList/OrderListSlice";

export const store = configureStore({
  reducer: {
    storeL: StoreReducer,
    user: UserReducer,
    notification: NotificationReducer,
    viewsubscription: SubscriptionReducer,
    // addsubscription: AddSubcriptionRe,
    cart: CartReducer,
    // order: OrderSlice,
    // orderL: OrderListSlice,
    manager: ManagerReducer,
  },
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: false,
    }),
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
