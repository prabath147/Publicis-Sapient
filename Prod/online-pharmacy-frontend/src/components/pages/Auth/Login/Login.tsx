import {
  Button,
  Container,
  Paper,
  PasswordInput,
  Text,
  TextInput,
  Title,
} from "@mantine/core";
import { useForm } from "@mantine/form";
import { showNotification } from "@mantine/notifications";
import { Link, useNavigate } from "react-router-dom";
import { useAppDispatch } from "../../../../app/hooks";
import AuthNavbar from "../../../ui/AuthNavbar/AuthNavbar";
import { loginAPI } from "./LoginAPI";
import { LoginRequest } from "./models";
import { LoginAction } from "./UserSlice";

export interface LoginFormProps {
  onSubmit?: (loginRequest: LoginRequest) => void;
}
export default function Login({ onSubmit }: LoginFormProps) {
  const navigate = useNavigate();
  const dispatch = useAppDispatch();

  const form = useForm<LoginRequest>({
    validateInputOnChange: true,

    initialValues: {
      username: "",
      password: "",
    },

    validate: {
      username: (value) =>
        value.length < 2 ? "Must have at least 2 letters" : null,
      password: (value) =>
        value.length < 2 ? "Must have at least 2 letters" : null,
    },
  });

  return (
    <>
      <AuthNavbar />
      <Container size={420} my={40}>
        <form
          onSubmit={form.onSubmit((values) => {
            if (onSubmit) {
              onSubmit(values);
            }

            loginAPI(values)
              .then((response) => {
                dispatch(LoginAction(response.data));
                showNotification({
                  message: "Logged in Successfully!",
                  color: "green",
                });
                navigate("/");
              })
              .catch((error) => {
                showNotification({
                  message: "Invalid Username Or Password",
                  color: "red",
                });
              });
          })}
        >
          <Title
            className="animate__animated animate__pulse"
            align="center"
            sx={() => ({ fontWeight: 900 })}
          >
            Log In
          </Title>
          <Text color="dimmed" size="sm" align="center" mt={5}>
            Do not have an account yet?{" "}
            <Link to="/register">Create account </Link>
          </Text>

          <Paper withBorder shadow="md" p={30} mt={30} radius="md">
            <TextInput
              mt={"md"}
              label="User Name"
              withAsterisk
              {...form.getInputProps("username")}
            />

            <PasswordInput
              mt={"md"}
              label="Password"
              withAsterisk
              {...form.getInputProps("password")}
            />

            <Text color="dimmed" size="sm" align="center">
              <Link to="/forgot-password-email">Forgot Password?</Link>
            </Text>
            <Button fullWidth mt="xl" type="submit">
              Sign in
            </Button>
          </Paper>
        </form>
      </Container>
    </>
  );
}
