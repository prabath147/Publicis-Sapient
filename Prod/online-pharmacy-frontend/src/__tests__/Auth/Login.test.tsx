import "@testing-library/jest-dom";
import { fireEvent, render, screen } from '@testing-library/react';
import { act } from 'react-dom/test-utils';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { store } from '../../app/store';
import Login from '../../components/pages/Auth/Login/Login';
import { LoginRequest, LoginResponse, User } from '../../components/pages/Auth/Login/models';
import mockAxios from "../../__mocks__/axios";

// setup
const onSubmit = jest.fn();

const MockLogin = () => {
  return (
    <Provider store={store}>
      <BrowserRouter>
        <Login onSubmit={onSubmit} />
      </BrowserRouter>
    </Provider>
  )
}


const loginOutput: LoginResponse = {
  email: "one@email.com",
  id: 1,
  username: "1234567890",
  refreshToken: "t1",
  token: "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwicm9sZSI6IlJPTEVfQURNSU4iLCJleHAiOjE1MTY5MzkwMjIsImlhdCI6MTUxNjIzOTAyMn0.lN42btTYcPU_3yBUC0Qh6qjWzpfagp3Av5MABEA15Y8"
}


it("component created", () => {
  const component = renderer.create(<MockLogin />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

const LoginInput: LoginRequest = {
  username: "ankit1",
  password: "1234",
};

describe("Login tests", () => {

  afterEach(() => {
    jest.resetAllMocks();
  });

  it("Check if form is rendered correctly", () => {
    render(<MockLogin />);

    expect(
      screen.getByRole("textbox", { name: /User Name/i })
    ).toBeInTheDocument();
    expect(screen.getByLabelText(/password/i)).toBeInTheDocument();
    expect(
      screen.getByRole("button", { name: /Sign in/i })
    ).toBeInTheDocument();
  });

  it("onSubmit should be called when the user inputs valid details", async () => {
    mockAxios.post.mockImplementationOnce(() =>
      Promise.resolve({ data: loginOutput })
    );

    const { getByLabelText, getByRole } = render(<MockLogin />);

    await act(async () => {
      fireEvent.change(getByRole("textbox", { name: /User Name/i }), {
        target: { value: "1" },
      });
      fireEvent.change(getByLabelText(/password/i), {
        target: { value: "1234" },
      });
    });

    await act(async () => {
      fireEvent.click(getByRole("button", { name: /Sign in/i }));
    });

    await act(async () => {
      fireEvent.change(getByRole("textbox", { name: /User Name/i }), {
        target: { value: "ankit1" },
      });
      fireEvent.change(getByLabelText(/password/i), {
        target: { value: "1234" },
      });
    });

    await act(async () => {
      fireEvent.click(getByRole("button", { name: /Sign in/i }));
    });

    expect(onSubmit).toBeCalledWith(LoginInput);
  });

  it("should handle errors", async () => {
    mockAxios.post.mockImplementationOnce(() => Promise.resolve());

    const { getByLabelText, getByRole } = render(<MockLogin />);

    await act(async () => {
      fireEvent.change(getByRole("textbox", { name: /User Name/i }), {
        target: { value: "ankit1" },
      });
      fireEvent.change(getByLabelText(/password/i), {
        target: { value: "1234" },
      });
    });

    const data: LoginRequest = {
      username: "ankit1",
      password: "1234",
    };

    await act(async () => {
      fireEvent.click(getByRole("button", { name: /Sign in/i }));
    });

    expect(onSubmit).toBeCalledWith(data);
  })

})

describe('User Slice tests', () => {
  const initialState: User = {
    id: -1,
    username: "",
    email: "",
    role: "",
  }

  const user: User = {
    email: "one@email.com",
    id: 1,
    username: "1234567890",
    role: "ROLE_ADMIN"
  }

  // TODO: update this
  // it("should handle initial state", () => {
  //     const actual: User = UserReducer(initialState, LoginAction(loginOutput))
  //     expect(actual).toEqual(user)

  //     const actual2: User = UserReducer(initialState, LogoutAction())
  //     expect(actual2).toEqual(initialState)
  // });
})


// function getUsername() {
//     return screen.getByRole('textbox', {
//         name: /username/i
//     });
// }

// function getPassword() {
//     return screen.getByLabelText(/password/i);
// }

// function clickSubmitButton() {
//     user.click(screen.getByRole('button', {
//         name: /sign in/i
//     }));
// }
