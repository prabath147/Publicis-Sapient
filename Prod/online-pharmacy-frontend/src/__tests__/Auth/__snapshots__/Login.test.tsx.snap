// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component created 1`] = `
Array [
  <div
    className="mantine-Container-root mantine-15pjuqq"
  >
    <div
      className="main-nav"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <div
          className="mantine-Image-root mantine-15po0m8"
          style={
            Object {
              "height": "auto",
              "width": "100%",
            }
          }
        >
          <figure
            className="mantine-11nhzn5 mantine-Image-figure"
          >
            <div
              className="mantine-qqmv3w mantine-Image-imageWrapper"
            >
              <img
                className="mantine-3y8yz3 mantine-Image-image"
                onError={[Function]}
                onLoad={[Function]}
                src="logo.jpg"
                style={
                  Object {
                    "height": "auto",
                    "objectFit": "cover",
                    "width": "100%",
                  }
                }
              />
            </div>
          </figure>
        </div>
      </a>
    </div>
  </div>,
  <div
    className="mantine-Container-root mantine-1tg98w8"
  >
    <form
      onSubmit={[Function]}
    >
      <h1
        className="mantine-Text-root mantine-Title-root animate__animated animate__pulse mantine-l5hqso"
      >
        Log In
      </h1>
      <div
        className="mantine-Text-root mantine-wtjg8l"
      >
        Do not have an account yet?
         
        <a
          href="/register"
          onClick={[Function]}
        >
          Create account 
        </a>
      </div>
      <div
        className="mantine-Paper-root mantine-16boms2"
      >
        <div
          className="mantine-InputWrapper-root mantine-TextInput-root mantine-1l2unfb"
        >
          <label
            className="mantine-InputWrapper-label mantine-TextInput-label mantine-ittua2"
            htmlFor="mantine-r0"
            id="mantine-r0-label"
          >
            User Name
            <span
              aria-hidden={true}
              className="mantine-u5apz8 mantine-InputWrapper-required mantine-TextInput-required"
            >
               *
            </span>
          </label>
          <div
            className="mantine-Input-wrapper mantine-TextInput-wrapper mantine-12sbrde"
          >
            <input
              aria-invalid={false}
              className="mantine-Input-input mantine-TextInput-input mantine-gqmpge"
              id="mantine-r0"
              onChange={[Function]}
              onFocus={[Function]}
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          className="mantine-InputWrapper-root mantine-PasswordInput-root mantine-1l2unfb"
        >
          <label
            className="mantine-InputWrapper-label mantine-PasswordInput-label mantine-ittua2"
            htmlFor="mantine-r1"
            id="mantine-r1-label"
          >
            Password
            <span
              aria-hidden={true}
              className="mantine-u5apz8 mantine-InputWrapper-required mantine-PasswordInput-required"
            >
               *
            </span>
          </label>
          <div
            className="mantine-Input-wrapper mantine-PasswordInput-wrapper mantine-12sbrde"
          >
            <div
              aria-invalid={false}
              className="mantine-PasswordInput-input mantine-Input-input mantine-PasswordInput-input mantine-6k8x11"
            >
              <input
                className="mantine-164y8ej mantine-PasswordInput-innerInput"
                id="mantine-r1"
                onChange={[Function]}
                onFocus={[Function]}
                type="password"
                value=""
              />
            </div>
            <div
              className="mantine-o3oqoy mantine-Input-rightSection mantine-PasswordInput-rightSection"
            >
              <button
                aria-hidden={true}
                className="mantine-UnstyledButton-root mantine-ActionIcon-root mantine-PasswordInput-visibilityToggle mantine-1dcetaa"
                disabled={false}
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                tabIndex={-1}
                type="button"
              >
                <svg
                  fill="none"
                  height={15}
                  viewBox="0 0 15 15"
                  width={15}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clipRule="evenodd"
                    d="M7.5 11C4.80285 11 2.52952 9.62184 1.09622 7.50001C2.52952 5.37816 4.80285 4 7.5 4C10.1971 4 12.4705 5.37816 13.9038 7.50001C12.4705 9.62183 10.1971 11 7.5 11ZM7.5 3C4.30786 3 1.65639 4.70638 0.0760002 7.23501C-0.0253338 7.39715 -0.0253334 7.60288 0.0760014 7.76501C1.65639 10.2936 4.30786 12 7.5 12C10.6921 12 13.3436 10.2936 14.924 7.76501C15.0253 7.60288 15.0253 7.39715 14.924 7.23501C13.3436 4.70638 10.6921 3 7.5 3ZM7.5 9.5C8.60457 9.5 9.5 8.60457 9.5 7.5C9.5 6.39543 8.60457 5.5 7.5 5.5C6.39543 5.5 5.5 6.39543 5.5 7.5C5.5 8.60457 6.39543 9.5 7.5 9.5Z"
                    fill="currentColor"
                    fillRule="evenodd"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
        <div
          className="mantine-Text-root mantine-1ggy16k"
        >
          <a
            href="/forgot-password-email"
            onClick={[Function]}
          >
            Forgot Password?
          </a>
        </div>
        <button
          className="mantine-UnstyledButton-root mantine-Button-root mantine-11wm0w1"
          data-button={true}
          type="submit"
        >
          <div
            className="mantine-3xbgk5 mantine-Button-inner"
          >
            <span
              className="mantine-qo1k2 mantine-Button-label"
              style={
                Object {
                  "textTransform": undefined,
                }
              }
            >
              Sign in
            </span>
          </div>
        </button>
      </div>
    </form>
  </div>,
]
`;
